File: java/src/programmers/타겟넘버.java
package programmers;

class 타겟넘버 {

	public int solution(int[] numbers, int target) {
		return dfs(numbers, 0, target, 0);
	}

	public int dfs(int[] numbers, int depth, int target, int result) {
		if (depth == numbers.length) {
			return (target == result) ? 1 : 0;
		}

		int plus = dfs(numbers, depth + 1, target, result + numbers[depth]);
		int minus = dfs(numbers, depth + 1, target, result - numbers[depth]);

		return plus + minus;
	}
}

File: python/programmers/타겟넘버.py
def solution(numbers, target):
    answer = 0
    bfs = [0]

    for num in numbers:
        temp = []
        for n in bfs:
            temp.append(n - num)
            temp.append(n + num)
        bfs = temp

    for n in bfs:
        if n == target:
            answer += 1
            
    return answer
